src/overlays/effects/ovl_Effect_Ss_Kakera/z_eff_ss_kakera.c

```diff
@@ -8,50 +8,349 @@
+        this->accel.y += D_8097EB14[index];
+    }
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E3C0.s")
+s32 func_8097E4F0(EffectSs* this, Vec3f* diff, f32 distance) {
+    f32 phi_f0;
+    s32 index = (this->regs[0] >> 5) & 3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E400.s")
+    if (index != 0) {
+        if (distance > 1.0f) {
+            phi_f0 = 1.0f / distance;
+        } else {
+            phi_f0 = 1.0f;
+        }
+        this->accel.x -= diff->x * D_8097EB30[index - 1] * phi_f0;
+        this->accel.z -= diff->z * D_8097EB30[index - 1] * phi_f0;
+    }
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E420.s")
+s32 func_8097E584(EffectSs* this, Vec3f* diff, f32 distance) {
+    f32 center;
+    f32 rand;
+    s32 index;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E4B0.s")
+    index = (this->regs[0] >> 7) & 0xF;
+    center = D_8097EB3C[index](distance, index);
+    rand = func_8097DE30(center, (this->regs[9] * center) / 1024.0f);
+    this->accel.x *= rand;
+    this->accel.y *= rand;
+    this->accel.z *= rand;
+    this->accel.x += rand * 0.01f;
+    this->accel.y += rand * 0.01f;
+    this->accel.z += rand * 0.01f;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E4F0.s")
+s32 func_8097E660(EffectSs* this, Vec3f* diff, f32 distance) {
+    this->accel.y += this->regs[1] / 256.0f;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E584.s")
+s32 func_8097E698(EffectSs* this) {
+    Vec3f diff;
+    f32 distance;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E660.s")
+    func_8097E34C(this);
+    diff.x = this->pos.x - this->vec.x;
+    diff.y = this->pos.y - this->vec.y;
+    diff.z = this->pos.z - this->vec.z;
+    distance = sqrtf(SQXYZ(diff));
+    if (distance > 1000.0f) {
+        return false;
+    }
+    if (this->regs[0] != 0) {
+        if (func_8097E420(this, &diff, distance) == 0) {
+            return false;
+        }
+        if (func_8097E4B0(this, &diff, distance) == 0) {
+            return false;
+        }
+        if (func_8097E4F0(this, &diff, distance) == 0) {
+            return false;
+        }
+        if (func_8097E584(this, &diff, distance) == 0) {
+            return false;
+        }
+    }
+    if (func_8097E660(this, &diff, distance) == 0) {
+        return false;
+    } else {
+        return true;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E698.s")
+void func_8097E7E0(EffectSs* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E7E0.s")
+    if (this->regs[8] == 0) {
+        if ((((this->regs[4] >> 4) & 1) * 0x10) == 0x10) {
```

Comment:
```suggestion
        if ((((this->regs[4] >> 4) & 1) << 4) == 0x10) {
```
Does this still match? This would match the format of the similar usages.

Reply:
I think it should be `(1 << 4)` on the right for consistency? Very weird thing to write...

Reply:
yep, changing both to shifts still matches. unfortunately the sane comparison here does not match.